### ISSUE TITLE
Using cross-encoders: fine-tuning required?

### ISSUE THREAD

**Issue by BramVanroy** [UKPLab/sentence-transformers#615]
In our research we want to get sentence similarity between sentences. For now we have always used the `model.encode()` functionality of the bi-encoder and then calculate the similarity manually (as seen [here](https://www.sbert.net/docs/usage/semantic_textual_similarity.html)). Because quality is important, we may want to move to cross-encoders, as you point out that it performs better (though more slowly) than bi-encoders.

From [the documentation](https://www.sbert.net/examples/applications/cross-encoder/README.html) it gives the impression that we can simply load any `transformers` model and use that to get similarity scores. However, I cannot imagine that that is correct: oftentimes the XXXForSequenceClassification heads are _not_ pretrained and if they are they are not trained on the similarity task. So I imagine that to successfully use cross-encoders, fine-tuning is necessary? As an additional question: I can see that there are a number of pretrained English CE models [available](https://www.sbert.net/docs/pretrained_cross-encoders.html). Do you also have multilingual models available?

**Comment by nreimers** [comment]
Hi @BramVanroy 

Yes, the cross-encoder has to be fine-tuned if used for semantic textual similarity. Without fine-tuning, you just get some random scores.

So far CE models are sadly only available for English. What works quite well for cross-encoders it to translate the training corpora using machine translation and to train them based on this.

**Comment by BramVanroy** [comment]
Thanks for the response! 

An additional question concerning the multilingual models (e.g. `xlm-r-bert-base-nli-stsb-mean-tokens`): from the paper I understand that you train (from-scratch, not finetuning?) XLM-R to imitate SBERT. My first question is: in the documentation it states "trained on 50+ languages" but I cannot find this in detail in the paper. Can you explain:

- whether that involves finetuning or training from-scratch?
- which dataset was used for this procedure? (I see that you used mSTS and translated it with Google translate, but even then you do not have 50 languages, hence my question)

I'm looking for detailed information as we are referring to sentence-transformers in an upcoming paper so I'd like to get the details right :-)

**Comment by nreimers** [comment]
Hi @BramVanroy 
Models were of course fine-tuned, not trained from scratch.

For the languages and datasets, see the paper:
https://arxiv.org/pdf/2004.09813.pdf

Section 3

**Comment by BramVanroy** [comment]
I read the paper again and now better understand what's going on. The teacher-student threw me off because the teacher is first finetuned on NLI, which I did not catch at a first read. Thanks!

**Comment by ratikapoor** [comment]
While using cross-encoder (CrossEncoder('distilroberta-base',num_labels=1)), I'm getting F1 score of 1 on the validation set with labels 0 and 1 after fine tuning. Although fine tuning is not happening properly. While inferencing it gives a constant score of 0.4118. Then also in logs, it shows 100 accuracy and F1 score.
CEBinaryClassificationEvaluator: Evaluating the model on fk-dev dataset after epoch 1:
Accuracy:           100.00      (Threshold: 0.4118)
F1:                 100.00      (Threshold: 0.4118)
Precision:          100.00
Recall:             100.00
Average Precision:  30.00

I'm not able to understand why it is not learning anything where I have 11k rows in the data with the target label equally distributed.

**Comment by BramVanroy** [comment]
@ratikapoor please post your own topic. Your question is unrelated to this topic. Don't add noise to unrelated questions.
